### ISSUE TITLE
Explore pre-trained models for Indic languages

### ISSUE THREAD

**Issue by evgenyrp** [mozilla/firefox-translations-training#1446]
As @ZJaume reported, we a bit struggle with quality for some Indic languages. There are some pretrained multilingual models available like https://github.com/AI4Bharat/IndicTrans2. We can explore using them as teachers or to generate synthetic data for the lowest-resource languages from this family. Many languages we train can potentially benefit from this.

**Comment by DavidVentura** [comment]
Do you consider this for other languages or families? I'm currently trying to make a model following some of the recipes and ideas from this repo, for Uyghur, by using HY-MT-2 as a teacher

NLLB is interesting but in my tests it was very brittle / hallucinated (at least on Uyghur)

**Comment by evgenyrp** [comment]
For LLMs, we do this for some languages https://mozilla.github.io/translations/docs/training/llm-synthetic/. Using them as a teacher alone requires too much compute (and requires heavy filtering with QE models) compared to training from scratch, but we might look more into it in the future. There's also licensing. On our [benchmarks](https://mozilla.github.io/translations/final-evals/), NLLB is not that good, at least the distilled version.

**Comment by ZJaume** [comment]
To track code progress of IndicTans2 model: https://github.com/mozilla/translations/pull/1449

**Comment by ZJaume** [comment]
Progress report on this.

| lang | model | flores COMET | Google diff |
| -- | -- | --: | --: |
| en-ur | indictrans2-1B | 83.59 | |
| en-ur | indictrans2-200M | 83.62 | |
| en-ur | indictrans distilled student | 83.16 | -0.07 |
| ur-en | indictrans2-1B | 87.69 | |
| ur-en  | indictrans2-200M | 87.05 | |
| ur-en | indictrans2 distilled student | 84.48 | -4.69 |
| ur-en | teacher pipeline | 86.22 |
| ur-en | student pipeline | 84.52 | -4.62 |
| en-mr | indictrans2-1B | 76.36 | |
| en-mr | indictrans2-200M | 76.27 | |
| en-mr | indictrans2 distilled student | 75.99 | -2.00 |
| en-mr | teacher pipeline | 73.93 | |
| mr-en | indictrans2-1B | 88.89 | |
| mr-en | indictrans2-200M | 87.64 | |
| mr-en | indictrans distilled student | 86.41 | -3.43 |
| mr-en | teacher pipeline | 87.40 | |
| mr-en | student pipeline | 85.86 | -3.98 |

 en-ur and en-mr went very well but ur-en did not, so I'm trying to see if there is something we can fix on that one. I'm also training mr-en, maybe we will see if it's a thing related to *->en direction.

**Comment by evgenyrp** [comment]
Do they report lower quality for xx-en in the paper?

**Comment by ZJaume** [comment]
They do report a chrF++ table comparing teachers (1B params) and students (200M params) and yes, differences (delta column is IT2 vs Dist-Seed) are bigger in *->en. I've updated the table above to add COMET scores for those students in my evaluations and it shows also that differences between indictrans2-1B and 200M are bigger in *->en. Our distillation for ur-en showed much more difference (-2.6 COMET vs -0.6), however it may not be related to indictrans2, our normal pipeline distillations also showed much bigger gap in *->en.

<img width="559" height="641" alt="Image" src="https://github.com/user-attachments/assets/1735ae28-1c20-4178-b773-3cea931ca956" />

**Comment by ZJaume** [comment]
The table has been updated with the latest mr-en distillation. It also shows the same trend of *->en = bigger gap, although not as bad as Urdu. But I think the distillations from IndicTrans2 are good in general.
